test/basic.ts

```diff
@@ -1,16 +1,18 @@
-import { createApp } from 'h3'
+import { createApp, eventHandler, fromNodeMiddleware, toNodeListener } from 'h3'
 import { listen } from 'listhen'
 import { createAuthMiddleware } from '../src'
 
 const app = createApp()
 
-app.useAsync(createAuthMiddleware({
+app.use(fromNodeMiddleware(createAuthMiddleware({
   provider: 'basic',
   sessionSecret: 'secret',
   username: 'test',
   password: 'test'
-}))
+})))
 
-app.useAsync(req => `Welcome ${req.auth.session.user}!`)
+// TODO: How to infer type here correctly?
+// @ts-ignore
+app.use(eventHandler(event => `Welcome ${event.req.auth.session.user}!`))
```

Comment:
The type inference breaks after updating to h3. I couldn't find info in the tests neither on how to properly set types. I think it'd be nice to showcase that in here *or* in the h3 tests, mainly https://github.com/unjs/h3/blob/main/test/integrations.test.ts#L27-L39

cc @pi0